sugarscape.py

```diff
@@ -154,24 +154,35 @@ def configureDiseases(self, numDiseases):
 
         diseaseEndowments = self.randomizeDiseaseEndowments(numDiseases)
         random.shuffle(self.agents)
-        diseases = []
         for i in range(numDiseases):
             diseaseID = self.generateDiseaseID()
             diseaseConfiguration = diseaseEndowments[i]
             newDisease = disease.Disease(diseaseID, diseaseConfiguration)
-            diseases.append(newDisease)
+            self.diseases.append(newDisease)
+        startingDiseases = self.configuration["startingDiseasesPerAgent"]
 
-        unplacedDisease = 0
-        for agent in self.agents:
-            for newDisease in diseases:
-                hammingDistance = agent.findNearestHammingDistanceInDisease(newDisease)["distance"]
-                if hammingDistance != 0:
+        if startingDiseases == [0, 0]:
+            diseases = self.diseases[:]
+            for agent in self.agents:
+                for newDisease in diseases:
+                    hammingDistance = agent.findNearestHammingDistanceInDisease(newDisease)["distance"]
+                    if hammingDistance != 0:
+                        agent.catchDisease(newDisease)
+                        diseases.remove(newDisease)
+                        break
+            if len(diseases) > 0 and ("all" in self.debug or "sugarscape" in self.debug):
+                print(f"Could not place {len(diseases)} diseases.")
+        else:
+            minStartingDiseases = startingDiseases[0]
+            maxStartingDiseases = startingDiseases[1]
+            currStartingDiseases = minStartingDiseases
+            for agent in self.agents:
+                agentDiseases = random.sample(self.diseases, currStartingDiseases)
```

Comment:
Does this ensure a roughly equal spread of _each_ disease created for the population, or will this result in a semi-random distribution of the diseases present? Is it possible (though unlikely) that a single disease could remain unplaced?

Reply:
The book specifies a completely random sample from the master disease list. It is possible that a single disease could remain unplaced in some configs, but it's extremely unlikely for the book examples since we have so many agents and 4/10 or 10/25 disease are given to each agent. I could modify this logic to use the pseudorandom uniform distribution we use for other endowments if you want.

Reply:
Yes, let's keep it consistent with the others. I think the random sample approach by the book is not worth reproducing exactly.